src/main/java/dev/openfeature/javasdk/Structure.java

```diff
@@ -0,0 +1,118 @@
+package dev.openfeature.javasdk;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import dev.openfeature.javasdk.internal.Pair;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@ToString
+@EqualsAndHashCode
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class Structure {
+
+    protected final Map<String, Pair<FlagValueType, Object>> attributes;
+
+    public Structure() {
+        this.attributes = new HashMap<>();
+    }
+
+    public Set<String> keySet() {
+        return this.attributes.keySet();
+    }
+
+    // getters
+    public Boolean getBooleanAttribute(String key) {
+        return getAttributeByType(key, FlagValueType.BOOLEAN);
+    }
+
+    public String getStringAttribute(String key) {
+        return getAttributeByType(key, FlagValueType.STRING);
+    }
+
+    public Integer getIntegerAttribute(String key) {
+        return getAttributeByType(key, FlagValueType.INTEGER);
+    }
+
+    public Double getDoubleAttribute(String key) {
+        return getAttributeByType(key, FlagValueType.DOUBLE);
+    }
+
+    /**
+     * Fetch date-time relevant key.
+     * 
+     * @param key feature key
+     * @return date time object.
+     * @throws java.time.format.DateTimeParseException if it's not a datetime
+     */
+    public ZonedDateTime getDatetimeAttribute(String key) {
+        String attr = getAttributeByType(key, FlagValueType.STRING);
+        if (attr == null) {
+            return null;
+        }
+        return ZonedDateTime.parse(attr, DateTimeFormatter.ISO_ZONED_DATE_TIME);
+    }
+
+    public Structure getStructureAttribute(String key) {
```

Comment:
We're also missing something like an array type.

Reply:
I think you are right.

Reply:
Added